tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,176 @@
+# Generates RegisterCodegenUnboxedKernels.cpp, UnboxingFunctions.h and UnboxingFunctions.cpp.
+import argparse
+import json
+import os
+from dataclasses import dataclass
+from tools.codegen.api.translate import translate
+from tools.codegen.api.types import CppSignatureGroup
+from tools.codegen.api.unboxing import convert_arguments
+from tools.codegen.context import method_with_native_function
+from tools.codegen.gen import parse_native_yaml
+from tools.codegen.model import NativeFunction, NativeFunctionsGroup, Variant
+from tools.codegen.utils import Target, FileManager, mapMaybe, make_file_manager
+from typing import Union, Sequence
+from typing_extensions import Literal
+
+
+# Generates UnboxingFunctions.h & UnboxingFunctions.cpp.
+@dataclass(frozen=True)
+class ComputeUnboxingFunctions:
+    target: Union[Literal[Target.DECLARATION], Literal[Target.DEFINITION]]
+
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=(Variant.method in f.variants), fallback_binding=f.manual_cpp_binding
```

Comment:
nit 1: the signature is only used in the `else` clause, so may as well wait until you're inside the `else` clause to generate it

nit 2: you don't care about `manual_cpp_binding` in the unboxing code and it has a default value anyway, so may as well not pass it in